code/modules/cm_marines/overwatch.dm

```diff
@@ -788,10 +794,14 @@
 		to_chat(usr, "[icon2html(src, usr)] [SPAN_WARNING("Transfer aborted. [transfer_marine] isn't wearing an ID.")]")
 		return
 
-	var/new_squad_name = tgui_input_list(usr, "Choose the marine's new squad", "Squad Selection", ROLES_SQUAD_ALL)
-	if(!new_squad_name || S != current_squad)
+	var/list/available_squads = list()
+	for(var/datum/squad/squad as anything in RoleAuthority.squads)
+		if(squad.active && !squad.locked && (squad.faction == faction) && (squad.name != "Root"))
+			available_squads += squad
+
+	var/datum/squad/new_squad = tgui_input_list(usr, "Choose the marine's new squad", "Squad Selection", available_squads)
+	if(!new_squad || S != current_squad)
```

Comment:
what's that current squad check for?

Reply:
This isn't a me one, pre-existing. It's to check if they've already been moved, mid move. Prevents it crashing by somehow happening twice.